spi/topology/src/main/java/quarks/topology/spi/graph/ConnectorStream.java

```diff
@@ -96,6 +89,35 @@ protected Graph graph() {
     }
 
     @Override
+    public <E extends Enum<E>> EnumMap<E,TStream<T>> split(Class<E> enumClass, Function<T, E> splitter) {
+
+        E[] es = enumClass.getEnumConstants();
+
+/*
+        List<TStream<T>> outputs = split(es.length, e -> IntStream.range(0, es.length).filter(i-> es[i].equals(splitter.apply(e)))
+            .mapToObj(i -> es[i].ordinal()).findAny().orElse(-1));
+*/
+
+        List<TStream<T>> outputs = split(es.length, new ToIntFunction<T>() {
+            @Override
+            public int applyAsInt(T input) {
+                for(int i = 0; i < es.length; i++){
```

Comment:
I was also going to comment on simplifying the fn.  And (then eliminating the need for the intermediate "es" variable) I think the returnMap construction could be simplified to:
```
for (E e : enumClass.getEnumConstants()) {
    returnMap.put(e, outputs.get(e.ordinal());
}
```